client/components/Main/CreateNewSubnavDropdown.jsx

```diff
@@ -3,43 +3,75 @@ import PropTypes from 'prop-types';
 
-    if (privileges[PRIVILEGES.PLANNING_MANAGEMENT]) {
-        items.push({
-            label: gettext('Planning Item'),
-            icon: 'icon-plus-sign icon--blue',
-            action: addPlanning,
-            id: 'create_planning',
-        });
-    }
+        const {addEvent, addPlanning, createPlanningOnly, privileges, dispatch} = this.props;
+        const items = [];
 
-    if (!createPlanningOnly && !!privileges[PRIVILEGES.EVENT_MANAGEMENT]) {
-        items.push({
-            label: gettext('Event'),
-            icon: 'icon-plus-sign icon--blue',
-            action: addEvent,
-            id: 'create_event',
-        });
-    }
+        if (privileges[PRIVILEGES.PLANNING_MANAGEMENT]) {
+            items.push({
+                label: gettext('Planning Item'),
+                icon: 'icon-plus-sign icon--blue',
+                action: addPlanning,
+                id: 'create_planning',
+            });
+        }
 
-    return (items.length === 0 ? null :
-        <Dropdown
-            icon="icon-plus-large"
-            label={gettext('Create new')}
-            items={items}
-            alignRight={true}
-            disableSelection={createPlanningOnly}
-            defaultAction={addPlanning}
-            tooltip={createPlanningOnly ? gettext('Create new planning item') : gettext('Create new item')} />
-    );
-};
+        if (!createPlanningOnly && !!privileges[PRIVILEGES.EVENT_MANAGEMENT]) {
+            items.push({
+                label: gettext('Event'),
+                icon: 'icon-plus-sign icon--blue',
+                action: addEvent,
+                id: 'create_event',
+            });
+
+            this.props.eventTemplates.forEach((template) => {
+                items.push({
+                    label: template.template_name,
+                    icon: 'icon-plus-sign icon--blue',
+                    action: () => dispatch(actions.main.createNew(ITEM_TYPE.EVENT, template.data)),
+                    id: template._id,
+                });
+            });
+        }
 
-CreateNewSubnavDropdown.propTypes = {
+        return (items.length === 0 ? null :
+            <Dropdown
+                icon="icon-plus-large"
+                label={gettext('Create new')}
+                items={items}
+                alignRight={true}
+                disableSelection={createPlanningOnly}
+                defaultAction={addPlanning}
+                tooltip={createPlanningOnly ? gettext('Create new planning item') : gettext('Create new item')}
+                scrollable
+                searchable
+            />
+        );
+    }
+}
+
+CreateNewSubnavDropdownFn.propTypes = {
     addEvent: PropTypes.func.isRequired,
     addPlanning: PropTypes.func.isRequired,
     createPlanningOnly: PropTypes.bool,
     privileges: PropTypes.object,
+    dispatch: PropTypes.func,
+    eventTemplates: PropTypes.array,
 };
+
+function mapStateToProps(state) {
+    return {
+        eventTemplates: state.events.eventTemplates,
```

Comment:
Please create a selector in `client/selectors/events.js` instead of directly accessing state.

Reply:
Why? Aren't selectors meant for maintaining the same reference when `map` or another reference changing is method is used? In this case, I'm not doing anything with the state and using a selector wouldn't have any impact. Or would it?

Reply:
One of the reasons why we use selectors when accessing state is that the usage of a state variable should not be dependent on where in the store that variable is. So if for whatever reason `eventTemplates` were to be moved to a different part of the store, or the name changed, or the structure changed, we would only have to change the selector that returns the `eventTemplates`.

If we were accessing the state directly, these references would have to be changed everywhere. So as a rule of thumb we use selectors when getting anything from the store.